tests/unit/amo/components/TestSearchForm.js

```diff
@@ -1,179 +1,394 @@
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+    });
+
+    it('does not display suggestions when there is no suggestion', () => {
+      wrapper = mountComponent({ suggestions: [] });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
+
+    it('displays 10 loading bars when suggestions are loading', () => {
+      const { store } = dispatchSignInActions();
+
+      store.dispatch(autocompleteStart({
+        errorHandlerId: createStubErrorHandler().id,
+        filters: { query: 'test' },
+      }));
+
+      wrapper = mountComponent(mapStateToProps(store.getState()));
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(LoadingText)).toHaveLength(10);
+    });
+
+    it('updates the state and push a new route when a suggestion is selected', () => {
+      const result = createFakeAutocompleteResult();
+      const { store } = dispatchAutocompleteResults({ results: [result] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      wrapper = mountComponent({
+        query: 'foo',
+        suggestions: autocompleteState.suggestions,
+      });
+      expect(wrapper.state('searchValue')).toEqual('foo');
+
+      wrapper.find('input').simulate('focus');
+      wrapper.find(Suggestion).simulate('click');
+      expect(wrapper.state('searchValue')).toEqual('');
+      sinon.assert.callCount(router.push, 1);
+      sinon.assert.calledWith(router.push, result.url);
+    });
+
+    it('ignores loading suggestions that are selected', () => {
+      const result = createFakeAutocompleteResult();
+      const { store } = dispatchAutocompleteResults({ results: [result] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      wrapper = mountComponent({ query: 'baz' });
+      expect(wrapper.state('searchValue')).toEqual('baz');
+
+      wrapper.instance().handleSuggestionSelected(createFakeEvent(), {
+        suggestion: {
+          ...autocompleteState.suggestions[0],
+          loading: true,
+        },
+      });
+      expect(wrapper.state('searchValue')).toEqual('baz');
+      sinon.assert.notCalled(router.push);
+    });
+
+    it('does not fetches suggestions when there is not value', () => {
+      const dispatch = sinon.spy();
+      wrapper = mountComponent({ dispatch });
+
+      wrapper.instance().handleSuggestionsFetchRequested({});
+      sinon.assert.notCalled(dispatch);
+    });
+
+    it('adds addonType to the filters used to fetch suggestions', () => {
+      const dispatch = sinon.spy();
+      wrapper = mountComponent({
+        query: 'ad',
+        addonType: ADDON_TYPE_EXTENSION,
+        dispatch,
+      });
+
+      wrapper.find('input').simulate('focus');
+
+      sinon.assert.callCount(dispatch, 1);
+      sinon.assert.calledWith(dispatch, autocompleteStart({
+        errorHandlerId: createStubErrorHandler().id,
```

Comment:
heh, this only works by coincidence. I think you need to move the variable to the upper scope so you can reference it. Example:

```diff
diff --git a/tests/unit/amo/components/TestSearchForm.js b/tests/unit/amo/components/TestSearchForm.js
index a4d678ef..2915f09a 100644
--- a/tests/unit/amo/components/TestSearchForm.js
+++ b/tests/unit/amo/components/TestSearchForm.js
@@ -32,6 +32,7 @@ import {
 describe(__filename, () => {
   const pathname = '/search/';
   const api = { clientApp: 'firefox', lang: 'de' };
+  let errorHandler;
   let router;
   let wrapper;
   let form;
@@ -46,7 +47,7 @@ describe(__filename, () => {
         i18n={getFakeI18nInst()}
         loadingSuggestions={false}
         suggestions={[]}
-        errorHandler={createStubErrorHandler()}
+        errorHandler={errorHandler}
         dispatch={() => {}}
         router={router}
         debounce={(callback) => (...args) => callback(...args)}
@@ -67,6 +68,7 @@ describe(__filename, () => {
       wrapper = mountComponent();
       form = wrapper.find('form');
       input = wrapper.find('input');
+      errorHandler = createStubErrorHandler();
     });
 
     it('renders a form', () => {
```

Reply:
indeed, thanks!